### ISSUE TITLE
API for exposing a public URL to a blob

### ISSUE THREAD

**Issue by ELLIOTTCABLE** [git-lfs/git-lfs#2372]
This is continuing a discussion [started by @maxnordlund](https://github.com/git-lfs/git-lfs/issues/1342#issuecomment-238663072) on #1342:

For things like GitHub Pages and READMEs to be able to include / reference `git-lfs` blobs, the API would first need to support a way for a client to *tell* the server that a particular resource should be publicly-accessible (which doesn't even begin to address how to *get* that publicly-accessible URL back out of the server; but that may be out-of-scope here.)

My personal rationale: This is extremely ideal solution to helping deal with the issue of including beautiful documentation in a project (screenshots, GIFs, and other large embeds) without ensuring that every single user who clones your project for the rest of history (at least, until you're forced to rewrite the history of the whole project, and break the *old* clones) is going to have to download the data behind that 10MB `.gif` you ill-advisedly included in the original README. :P

**Comment by ttaylorr** [comment]
Hi @ELLIOTTCABLE, thanks for opening this issue. I think there are two points of discussion:

1. Should the LFS API specification include a section regarding public blob access (and blob permissions)?
2. Should this be a feature of GitHub Pages (or more generally, other consumers that use the raw.githubusercontent.com service).

I'm hesitant to recommend moving forward with adding this to the LFS specification, since I think this necessitates a permission scheme in the LFS API, which I'd like to avoid the complexity of.

Teach raw.githubusercontent.com to "upgrade" the LFS pointer file into an LFS blob would be interesting, and I can certainly pass the suggestion along. 

@technoweenie any thoughts?

**Comment by ELLIOTTCABLE** [comment]
Can we bring it down from a full ‘permission scheme’ to a simple boolean? Seems like premature abstraction, second only to premature optimization and all that. 😋

**Comment by technoweenie** [comment]
I don't think this is a concern for the LFS open source client or API specification. Adding a `public` boolean to the LFS API won't make it any easier to support LFS objects in a dependent service (CI services, GH Pages, etc).

GitHub will already let you host images in your repositories that are stored with LFS. But GitHub will also charge you for LFS bandwidth. This is something I'd really like to solve, while continuing to protect us against abuse. That's also separate from the LFS client and specification, so it's not something we'll be solving here with a PR.
